ui/app/components/ui/sender-to-recipient/sender-to-recipient.component.js

```diff
@@ -1,211 +1,265 @@
+  recipientEns,
+  onRecipientClick,
+  onSenderClick,
+  recipientAddress,
+  variant = DEFAULT_VARIANT,
+}) {
+  const { dappIsAuthorizedForAccount, accountIsCurrentlySelected } = useDappAuthorizationCheck(senderAddress)
+
+  const showWarning = dappIsAuthorizedForAccount && !accountIsCurrentlySelected
+
+  const [senderAddressCopied, setSenderAddressCopied] = useState(false)
+  const t = useContext(I18nContext)
+  const checksummedSenderAddress = checksumAddress(senderAddress)
+  const checksummedRecipientAddress = checksumAddress(recipientAddress)
+
+  return (
+    <div className={classnames('sender-to-recipient', variantHash[variant])}>
       <div
-        className="sender-to-recipient__party sender-to-recipient__party--recipient sender-to-recipient__party--recipient-with-address"
+        className={classnames('sender-to-recipient__party sender-to-recipient__party--sender')}
         onClick={() => {
-          copyToClipboard(checksummedRecipientAddress)
-          if (onRecipientClick) {
-            onRecipientClick()
+          setSenderAddressCopied(true)
+          copyToClipboard(checksummedSenderAddress)
+          if (onSenderClick) {
+            onSenderClick()
           }
         }}
       >
-        { this.renderRecipientIdenticon() }
-        <Tooltip
-          position="bottom"
-          html={
-            this.state.senderAddressCopied
-              ? <p>{t('copiedExclamation')}</p>
-              : (addressOnly && !recipientNickname && !recipientEns)
-                ? <p>{t('copyAddress')}</p>
-                : (
-                  <p>
-                    {shortenAddress(checksummedRecipientAddress)}<br />
-                    {t('copyAddress')}
-                  </p>
-                )
-          }
-          wrapperClassName="sender-to-recipient__tooltip-wrapper"
-          containerClassName="sender-to-recipient__tooltip-container"
-        >
-          <div className="sender-to-recipient__name">
-            <span>{ addressOnly ? `${t('to')}: ` : '' }</span>
-            {
-              addressOnly
-                ? (recipientNickname || recipientEns || checksummedRecipientAddress)
-                : (recipientNickname || recipientEns || recipientName || this.context.t('newContract'))
-            }
+        {!addressOnly && (
+          <div className="sender-to-recipient__sender-icon">
+            <Identicon
+              address={checksumAddress(senderAddress)}
+              diameter={24}
+            />
           </div>
-        </Tooltip>
-      </div>
-    )
-  }
-
-  renderRecipientWithoutAddress () {
-    return (
-      <div className="sender-to-recipient__party sender-to-recipient__party--recipient">
-        { !this.props.addressOnly && <i className="fa fa-file-text-o" /> }
-        <div className="sender-to-recipient__name">
-          { this.context.t('newContract') }
-        </div>
+        )}
+        <SenderAddress
+          checksummedRecipientAddress={checksummedRecipientAddress}
```

Comment:
I think this was supposed to be `checksummedSenderAddress` instead of `checksummedRecipientAddress`

```suggestion
          checksummedSenderAddress={checksummedSenderAddress}
```